src/game/client/components/menus.cpp

```diff
@@ -1783,8 +1586,8 @@ void CMenus::RenderMenu(CUIRect Screen)
 			Box.HSplitBottom(ButtonHeight*2.0f, 0, &Box);
 			Box.HSplitTop(ButtonHeight, &EditBox, &Box);
 
-			static float s_OffsetName = 0.0f;
-			DoEditBoxOptionUTF8(Config()->m_PlayerName, Config()->m_PlayerName, sizeof(Config()->m_PlayerName), MAX_NAME_LENGTH, &EditBox, Localize("Nickname"), ButtonWidth, &s_OffsetName);
+			static CLineInput s_PlayerNameInput(Config()->m_PlayerName, sizeof(Config()->m_PlayerName), MAX_NAME_LENGTH);
```

Comment:
What happens when such a string gets manipulated outside the editbox? Like another editbox like here, console, bind etc?

Reply:
The second commit changes it so now UpdateStrData is always called before processing input to handle external changes to the buffer. This is already part of the large PR but was not in this particular commit.